ci/infra/testrunner/skuba/skuba.py

```diff
@@ -104,12 +105,30 @@ def join_nodes(self, masters=None, workers=None):
         if workers is None:
             workers = self.platform.get_num_nodes("worker")
 
-        for n in range(1, masters):
-            self.node_join("master", n)
-
-        for n in range(0, workers):
-            self.node_join("worker", n)
-
+        nodes = [("master", n) for n in range(1, masters)] + \
+                [("worker", n) for n in range(0, workers)] 
+        for r, n in nodes:
```

Comment:
can we expand this to read `role, node_number` instead? Makes it clearer to know what we are doing :)